### DISCUSSION TITLE
how can MSPM0G3507 parameters Control? (CCS)

### DISCUSSION THREAD

**Discussion by nooby856** [community/community#142482] [Programming Help]
### Select Topic Area

Question

### Body

#include <ti/iqmath/include/IQmathLib.h>
#include "ti_msp_dl_config.h"

unsigned int TMRs = 0, DACs = 0;
signed int DIRs = 1;

_iq18 Angs, Outs;

int main(void)
{
    // 시스템 초기화
    SYSCFG_DL_init();

    // 타이머 인터럽트 활성화
    NVIC_EnableIRQ(myTMR_INST_INT_IRQN);
    // 저전력 모드에서 CPU 대기 설정
    DL_SYSCTL_enableSleepOnExit();
    // 타이머 시작
    DL_TimerA_startCounter(myTMR_INST);

    while (1)
    {
        __WFI(); // 대기
    }
}

void myTMR_INST_IRQHandler(void)
{
    // 타이머 값 증가
    TMRs += 1;

    // ADC 변환 시작 및 종료
    DL_ADC12_startConversion(myADC_INST);
    DL_ADC12_stopConversion(myADC_INST);

    // A1_0 핀에서 가변저항 값 읽기 (초기값 설정 추가)
    unsigned int A1_0 = 0;
    A1_0 = DL_ADC12_getMemResult(myADC_INST, myADC_ADCMEM_0);

    // 가변저항 값에 따른 파형 생성
    if (A1_0 < 2000) // 톱니파
    {
        DACs = TMRs & 0xFFF;
    }
    else if (A1_0 >= 2000 && A1_0 < 4000) // 삼각파
    {
        if (TMRs >= 0xFFF)
        {
            DIRs = -1;
        }
        else if (TMRs <= 0)
        {
            DIRs = 1;
        }
        TMRs += DIRs;
        DACs = TMRs;
    }
    else // 사인파
    {
        Angs += _IQ18(0.01);
        if (Angs >= _IQ18(6.283185307179586476925286766559))
        {
            Angs = _IQ18(0);
        }
        Outs = _IQ18sin(Angs) + _IQ18(1);
        DACs = Outs >> 7;
    }

    // DAC 출력
    DL_DAC12_output12(DAC0, DACs);
}

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @nooby856 ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
